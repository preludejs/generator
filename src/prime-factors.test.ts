import * as G from './index.js'

test.each([
  0,
  42,
  100,
  Number.MAX_SAFE_INTEGER
])('prime factors of %i', _ => {
  expect(G.array(G.primeFactors(_))).toMatchSnapshot()
})
