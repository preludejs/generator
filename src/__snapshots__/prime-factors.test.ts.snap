// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prime factors of 0 1`] = `Array []`;

exports[`prime factors of 42 1`] = `
Array [
  Array [
    2,
    1,
  ],
  Array [
    3,
    1,
  ],
  Array [
    7,
    1,
  ],
]
`;

exports[`prime factors of 100 1`] = `
Array [
  Array [
    2,
    2,
  ],
  Array [
    5,
    2,
  ],
]
`;

exports[`prime factors of 9007199254740991 1`] = `
Array [
  Array [
    6361,
    1,
  ],
  Array [
    69431,
    1,
  ],
  Array [
    20394401,
    1,
  ],
]
`;
